addon/components/column-chart/component.js

```diff
@@ -279,13 +279,16 @@ export default BaseChartComponent.extend({
 
     getLegendables(chart) {
         let legendables = [];
-        const formatter = this.get('xAxis.formatter') || (value => value);
         const data = this.get('data');
 
         this.get('series').forEach((series, i) => {
             if (!series.alert) {
-                let legendable = {};
+                let legendable = {
```

Comment:
Why are we defining the two different properties of the legendable in two different places? I just feel like this is separating logically similar operations. Does this fix an error that you're seeing?

Reply:
Yes, this PR fixes two issues.

Setting the title property here is necessary so that the legend can be shown even when all groups are empty. The `series` may be defined but there are no rectangles rendered. We can display the legend title without examining any of the rectangles. I think you're right though that they don't need to be separate- it seems like the color could also be moved up to the top here because it doesn't actually depend on the rectangles?

The other issue was that the rectangle filter was comparing the formatted value to the raw numeric value, so it always failed if the formatter returned anything but the raw value.

Reply:
So, the color can't be moved up because of the potential for hatching. The code below determines the color based on the `fill` variable, which needs to be checked on the actual rectangle itself since the hatching is done separately from the `colors` parameter.

Reply:
Also, how can we show anything if there aren't any rectangles? What color will the rectangle be? It doesn't make sense to show titles for rectangles that don't exist on the chart, it's just confusing.

Reply:
Oh right, forgot about the hatching being based on `this` which looks like it's the equivalent of `d` in that `filter` call. But, looks like the hatching is actually driven by a value on the `series` anyways right? So we could determine the legend fill based on that? (And there's the pos vs. neg hatch which I don't think will be reflected in the legend currently. Might be worth extracting that logic to compute a fill based on a `series`?)

A chart could have a legend if there are no rectangles. For example, if I chart answered calls, and the value is zero in every time interval, I still want to see what it is I'm charting (number of answered calls over time). The line chart renders a legend when there are no values.